builder/common/step_key_pair_test.go

```diff
@@ -0,0 +1,84 @@
+package common
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/hashicorp/packer-plugin-sdk/communicator"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+type mockEC2KeyPairConn struct {
+	ec2iface.EC2API
+
+	CreateKeyPairCount int
+	CreateKeyPairArgs  []ec2.CreateKeyPairInput
+
+	lock sync.Mutex
+}
+
+func (m *mockEC2KeyPairConn) CreateKeyPair(keyPairInput *ec2.CreateKeyPairInput) (*ec2.CreateKeyPairOutput, error) {
+	m.lock.Lock()
+	m.CreateKeyPairCount++
+	m.CreateKeyPairArgs = append(m.CreateKeyPairArgs, *keyPairInput)
+	m.lock.Unlock()
+	keyMaterial := "I'm a cool key"
+	output := &ec2.CreateKeyPairOutput{
+		KeyMaterial: &keyMaterial,
+	}
+	return output, nil
+}
+
+func getKeyPairMockConn() ec2iface.EC2API {
+	return &mockEC2KeyPairConn{}
+}
+
+func keyPairState() multistep.StateBag {
+	state := new(multistep.BasicStateBag)
+	state.Put("ui", &packersdk.BasicUi{
+		Reader: new(bytes.Buffer),
+		Writer: new(bytes.Buffer),
+	})
+	conn := getKeyPairMockConn()
+	state.Put("ec2", conn)
+	state.Put("region", "us-east-1")
+	return state
+}
+
+//Add "AndType" to the name of this test
+func TestStepKeyPair_callsCreateKeyPairOnceWithName(t *testing.T) {
```

Comment:
I would add an additional set of tests and rename this one a bit, as I believe you want to test that the correct inputs are being added to the CreateKeyPairInput type. 

```suggestion
func TestStepKeyPair_withDefault(t *testing.T) {
```
This should assert that the temp key name matches and that the default keytype is rsa

Then I would add an additional test maybe called `func TestStepKeyPair_withKeyType` that does the same as this test but checks that the key type is set to the expected value.